sdk/core/azure-core/vcpkg/portfile.cmake

```diff
@@ -21,6 +21,7 @@ vcpkg_cmake_configure(
         ${FEATURE_OPTIONS}
         -DWARNINGS_AS_ERRORS=OFF
         -DBUILD_TESTING=OFF
+        -DNO_AUTOMATIC_TRANSPORT_BUILD=ON
```

Comment:
If we must keep this flag to solve the issue, we'd want to document it here:
https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#cmake-build-options

Reply:
Do we have to document it? If we document it, it introduces complexity for the customer, as you said (we could though, it is not that big a complexity).
If we don't document it, everything works exactly the same.

Reply:
I mean, one could say, we cmake `SET()` a bunch of cmake variables on our cmakefiles, but we don't document them all as build flags.

Reply:
FWIW, I'm with Anton here. We document the things that we believe customers are likely to want to change and the reasons why they would want to change them.

And once we document them, they become fixed aspects of our build system and changes to them become breaking changes.

Right now, this flag (which exists to align the azure-sdk-for-cpp build environment with the requirements of vcpkg) doesn't meet that bar, IMHO, it is an implementation detail associated with the portfile.cmake and vcpkg.json file.

Note that this *does* apply to the contents of the vcpkg.json file. The features included in the file are a part of the public contract for the azure-core-cpp package. 

Changes to the features in the vcpkg package should be scrutinized for breaking changes whenever they are made.